spec/lib/vanagon/component_spec.rb

```diff
@@ -242,6 +242,49 @@
     end
   end
 
+  describe '#force_version' do
+    let(:source) {
+      allow(File).to receive(:realpath).and_return('/this/is/a/test')
+      Vanagon::Component::Source::Git.new('git://github.com/puppetlabs/facter', workdir: '/this/is/a/test')
+    }
+
+    let(:component) {
+      Vanagon::Component.new('force-version-test', {}, {}).tap do |comp|
+        comp.url = 'git://github.com/puppetlabs/facter'
+        comp.source = source
+      end
+    }
+
+    let(:versioned_component) {
+      Vanagon::Component.new('force-version-test', {}, {}).tap do |comp|
+        comp.version = '1.2.3'
+      end
+    }
+
+    it 'returns a version if it\'s already set' do
+      expect(versioned_component).not_to receive(:get_source)
+      expect(versioned_component.force_version).to eq('1.2.3')
+    end
+
+    it 'fetches sources if no version is set' do
+      clone = double(Git::Base)
+      allow(::Git).to receive(:clone).and_return(clone)
+      allow(clone).to receive(:describe).and_return('1.2.3')
```

Comment:
I feel like testing with something other than `1.2.3` would make it clearer that this is getting a different version than the one explicitly set in the test above.